### DISCUSSION TITLE
Support for in-library extensions

### DISCUSSION THREAD

**Discussion by kibertoad** [biomejs/biome#3270] [Ideas]
It would be great if it would be possible to publish a library with Biome configs, that could extend from each other, e. g:

biome-config (npm module):
configs (folder within module) ->
  biome-base.json
  biome-esm.json
  biome-backend.json
  biome-frontend.json

It is already possible to reference in-library configs like this:
```
	"extends": ["./node_modules/@kibertoad/biome-config/configs/biome-backend.json"],
```

However, it doesn't seem possible to extend from a rule that is colocated in the same directory within a package.

E. g. I've tried the following within `biome-backend.json` (located in an NPM module):

`"extends": ["./node_modules/@kibertoad/biome-config/configs/biome-base.json"],`
`"extends": ["./biome-base.json"],`
`"extends": ["biome-base.json"],`

None of them seem to work, base seems to be silently ignored without an error.

**Comment by ematipico** [comment]
I think this is similar to https://github.com/biomejs/biome/discussions/3122

I am concerned about adding more complexity to the resolution of the configuration, because this could trigger a number of recursions which are theoretically infinite. And we know what happened to eslint (aka flat config).

I understand the use case a bit (trying to make things generic and reusable, I suppose). Isn't there another alternative? Can't you do what you need with the resources that you have?

**Reply by kibertoad** [reply]
@ematipico How else can be this achieved without hard-to-maintain copy-pasting of rules?

Generally there are at least two sets of rules - for BE and for FE, and a shared base. How can it be done without copy-paste?

**Reply by ematipico** [reply]
Can't you export all the configuration files from the package, and then do something like this?

```json
{
	"extends": ["package-name/base", "pacakge-name/backend"]
}
```

**Reply by kibertoad** [reply]
Good point! Will give it a try, might work
